### PR TITLE
AZP/RELEASE: Scheduled DRP testing

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
## What
Add scheduled testing for the DRP release environment.

## Why ?
https://jirasw.nvidia.com/browse/HPCINFRA-2291 (internal link)

## How ?
Scheduled builds to create artifacts but not publish them.

Emergency release procedure:
1. Disable the main release pipeline.
2. Proceed as usual.

The DRP release pipeline will check the status of the main release pipeline.
Only if main is disabled will it run the full release cycle including the artifact's publishing.

### CODE REVIEW

**Comment by Alexey-Rivkin** [comment]
Test build with the main pipeline disabled:
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=84325&view=logs&j=6b27925f-05e1-5266-7798-9180c853e91e&t=129f6d2d-8a61-5ad7-a74a-1a0612b716fd&l=92

File: buildlib/az-helpers.sh
Comment (yosefe) [inline buildlib/az-helpers.sh:227]: wrong indent

File: buildlib/az-helpers.sh
Comment (yosefe) [inline buildlib/az-helpers.sh:221]: shouldn't we check *DRP match under this branch?

File: buildlib/azure-pipelines-release-drp.yml
Comment (yosefe) [inline buildlib/azure-pipelines-release-drp.yml:19]: remove commented out line?

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:221]: The scheduler is only set for the DRP pipeline, but it's better to be on the safe side.

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:227]: It was a bit more readable that way😪

File: buildlib/azure-pipelines-release-drp.yml
Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-release-drp.yml]: This was supposed to be a temp workaround, but as the PDC mirror is still down, let's keep using the main repo.

**Review by Alexey-Rivkin — COMMENTED** [review]
Applied all remarks

File: buildlib/az-helpers.sh
Comment (yosefe) [reply buildlib/az-helpers.sh:227]: IMO it's less readable because it does not comply with the if nesting level

File: buildlib/azure-pipelines-release-drp.yml
Comment (yosefe) [inline buildlib/azure-pipelines-release-drp.yml:19]: pls remove trailing space

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-release-drp.yml]: sorry

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh]: "then" was definitely wrongly indented, thx

**Review by yosefe — DISMISSED** [review]

**Comment by yosefe** [comment]
@Alexey-Rivkin pls squash

File: buildlib/az-helpers.sh
Comment (dpressle) [inline buildlib/az-helpers.sh:229]: I would call it from check_main_pipeline_status function as it is the one that needs it, in your implementation anyone who calls this function needs to call the install function before which seems redandent and confusing

File: buildlib/azure-pipelines-release-drp.yml
Comment (dpressle) [inline buildlib/azure-pipelines-release-drp.yml:10]: we probably should not run it nightly, once a week i smore then enough

**Review by dpressle — CHANGES_REQUESTED** [review]

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:229]: My first take was the same. The output from both functions (install_az_tools, check_main_pipeline_status) got mixed up in Azure, causing the "launch" variable to be assigned a long, meaningless value.

Also, the install function is called only for the DRP pipeline.

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:220]: My purpose was not to refactor the code but to introduce minimal changes while following the existing syntax.

File: buildlib/azure-pipelines-release-drp.yml
Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-release-drp.yml:10]: I'd better check it daily to catch it faster and make it easier to find the root cause.

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:220]: As for the single vs. double brackets usage:
Single brackets do not support pattern matching with wildcards. That's why I had to make it double for the second condition check.

Comment (dpressle) [reply buildlib/az-helpers.sh:220]: I think you should learn what is the differance between bash and posix, this should work fine:
if [[ $build_reason = "Schedule" && $BUILD_DEFINITIONNAME = *"DRP" ]]

File: buildlib/azure-pipelines-release-drp.yml
Comment (dpressle) [reply buildlib/azure-pipelines-release-drp.yml:10]: This is way too often, and since these processes are sensitive to false positive failures i dont want you (or me if you are not here) to have to monitor and debug on daily basis.
We also need to talk about who/what is monitoring this process this i snot something that only you should be aware of.

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-release-drp.yml]: Ok

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh]: Ok

**Comment by dpressle** [comment]
Let's not merge it yet, we still dont have pdc harbor working and i dont want to push a fix soon just for harbor url

File: buildlib/azure-pipelines-release-drp.yml
Comment (dpressle) [inline buildlib/azure-pipelines-release-drp.yml:19]: Lets wait for the PDC harbor to work (next week) before we merge it

**Review by dpressle — CHANGES_REQUESTED** [review]

**Review by dpressle — APPROVED** [review]

File: buildlib/azure-pipelines-release-drp.yml
Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-release-drp.yml]: Updated

**Review by dpressle — APPROVED** [review]

File: buildlib/azure-pipelines-release-drp.yml
Comment (yosefe) [inline buildlib/azure-pipelines-release-drp.yml:85]: why is this needed?

File: buildlib/az-helpers.sh
Comment (yosefe) [reply buildlib/az-helpers.sh:229]: Do we really have to install tools as part of test pipeline? Can't we do it as part of system setup?

File: buildlib/azure-pipelines-release-drp.yml
Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-release-drp.yml:85]: Authentication is required for `az` CLI down the line.
The actual authentication token is stored securely in Azure and injected at runtime, without the risk of exposure in logs or the pipeline YAML.

File: buildlib/az-helpers.sh
Comment (Alexey-Rivkin) [reply buildlib/az-helpers.sh:229]: Removed `az` tools installation from CI and updated the documentation [(internal link)](https://confluence.nvidia.com/display/NSWX/DRP%3A+UCX+release).

**Review by dpressle — APPROVED** [review]

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**
